Update nucleotide-count description (#1719)
exercism/problem-specifications: exercises/nucleotide-count/description.md

> Most people inherit not only their socioeconomic class from their parents, but also DNA (deoxyribonucleic acid), a set of chemical instructions that influence how their bodies are constructed!

Author: ```suggestion
Most people inherit not only their socioeconomic class from their parents, but also DNA, a set of chemical instructions that influence how their bodies are constructed!
```

> DNA is a long chain of other chemicals and the most important are the four nucleotides, adenine, cytosine, guanine and thymine. A single DNA chain can contain billions of these four nucleotides and the order in which they occur is important!

Author: ```suggestion

> Note: You do not need to understand anything about nucleotides or DNA to complete this exercise.

DNA is a long chain of other chemicals and the most important are the four nucleotides, adenine, cytosine, guanine and thymine. A single DNA chain can contain billions of these four nucleotides and the order in which they occur is important!
```

We call the order of these nucleotides in a bit of DNA a "DNA sequence".

We represent a DNA sequence as an ordered collection of these four nucleotides and a common way to do that is with a string of characters such as "ATTACG" for a DNA sequence of 6 nucleotides.
'A' for adenine, 'C' for cytosine, 'G' for guanine, and 'T' for thymine.

Given a string representing a DNA sequence, count how many of each nucleotide is present.
If the string contains characters that aren't A, C, G, or T then it is invalid and you should signal an error.

For example:

```
"GATTACA" -> 'A': 3, 'C': 1, 'G': 1, 'T': 2
"INVALID" -> error
```

> 
> You do not need to understand anything about nucleotides or DNA to complete this exercise.

Author: ```suggestion
```

> Most people inherit not only their socioeconomic class from their parents, but also DNA, a set of chemical instructions that influence how their bodies are constructed!

Reviewer: ```suggestion
Each of us inherits from our biological mother and father a set of chemical instructions known as DNA that influences how are bodies are constructed.
```
@cmcaine, just a suggestion.

Reviewer 2: I like the suggestion!

Author: If I can't think of something I like better then I'll take this but replace mother and father with parents 👍

Author: ```suggestion
Each of us inherits from our biological parents a set of chemical instructions known as DNA that influence how our bodies are constructed. All known life depends on DNA or something very similar!
```

I wanted it to still have a bit of a hook, and this'll do.

Reviewer: > or something very similar!

What would that be?  I generally like your suggested change but this last part left me wondering what that similar thing could be.  Maybe it doesn't need to be stated.

Author: RNA can replicate on its own under certain conditions. Depends on what you define life as, really. Maybe best to leave it out to avoid distracting people!

> Most people inherit not only their socioeconomic class from their parents, but also DNA (deoxyribonucleic acid), a set of chemical instructions that influence how their bodies are constructed!

Reviewer 2: I'm not sure about the "Most people inherit not only their socioeconomic class from their parents" introduction, as I can imagine this could be off-putting to some people.

Author: It puts me off society, too ;)

I'll see if I can come up with a good alternative.

Reviewer 3: Maybe remove the not only part and say we inherit heriditary aspects through DNA?

> Each of us inherits from our biological parents a set of chemical instructions known as DNA that influence how our bodies are constructed. All known life depends on DNA or something very similar!

Author: ```suggestion
Each of us inherits from our biological parents a set of chemical instructions known as DNA that influence how our bodies are constructed. All known life depends on DNA!
```

> DNA is a long chain of other chemicals and the most important are the four nucleotides, Adenine, Cytosine, Guanine and Thymine. A single DNA chemical can contain billions of these four nucleotides and the order in which they occur is important!

Author: Maybe I should say molecule rather than chemical? Not sure which is clearer.

Reviewer 4: It's not considered one molecule, is it? (I don't know anything about biology)

Author: It is a molecule :)

Author: > A molecule is an electrically neutral group of two or more atoms held together by chemical bonds.

DNA is held together by covalent and hydrogen bonds. Anyway, as someone who didn't know that, which is more confusing, saying "DNA molecule" or "DNA chemical"

Ah, I can just call it a "DNA chain" again instead.

We call the order of these nucleotides in a bit of DNA a "DNA sequence".

We represent a DNA sequence as an ordered collection of these four nucleotides and a common way to do that is with a string of characters such as "ATTACA" for a DNA sequence of 6 nucleotides.

Given a string representing a DNA sequence, count how many of each nucleotide is present.

> If the string contains characters that aren't A, C, G, or T then it is invalid and you should throw an error.

Reviewer 2: > If the string contains characters that aren't A, C, G, or T 

This is defining the character encoding, but to me it now feels separate for the actual nucleotides. I liked the previous version a bit better, where the nucleotides and their encoding are introduced jointly.

> 'A' for adenine, 'C' for cytosine, 'G' for guanine, and 'T' thymine

For example:

```
"GATTACA" -> 'G': 1, 'A': 3, 'T': 2, 'C': 1
"INVALID" -> error

> ```

Reviewer 2: I'd argue that "you should throw an error" is a term that does not apply to all tracks. Some tracks will return an error code or even an error type.  My suggestion would be to either rephrase a bit.

Reviewer 3: Yeah. But everything before it is good.

Is invalid will hint at "special case".

Author: Changed to read "signal an error"

> Given a society that determines life outcomes by inheritance, [rebel](https://www.nonviolent-conflict.org/).

Reviewer 5: I'm sympathetic to the sentiment, but tongue-in-cheek though it is, this might actually get is in legal trouble in some jurisdictions.

Reviewer 2: I agree that this should probably not be part of the exercise. It is a good cause, but not something I think well suited towards an exercise.

> You do not need to understand anything about nucleotides or DNA to complete this exercise.

Reviewer 3: Second paragraph, right after "yoooo DNA is cool" but before "it's made up of these amino acids".

 I would put it in a blockquote until @taiyab gives us alert like tools.

What do you think?

> You do not need to understand anything about nucleotides or DNA to complete this exercise.

Author: Agree about the blockquote, nice idea! I define what DNA is made from in the same paragraph (different line, tho), and I don't really want to break it up, but I guess it doesn't look too bad?

Most people inherit not only their socioeconomic class from their parents, but also DNA (deoxyribonucleic acid), a set of chemical instructions that influence how their bodies are constructed!

> Note: you do not need to understand anything about DNA to complete this exercise.

DNA is a long chain of other chemicals and the most important are the four nucleotides, adenine, cytosine, guanine and thymine. A single DNA chain can contain billions of these four nucleotides and the order in which they occur is important! We call the order of these nucleotides in a bit of DNA a "DNA sequence".

We represent a DNA sequence as an ordered collection of these four nucleotides and a common way to do that is with a string of characters such as "ATTACG" for a DNA sequence of 6 nucleotides. 'A' for adenine, 'C' for cytosine, 'G' for guanine, and 'T' for thymine.

Given a string representing a DNA sequence, count how many of each nucleotide is present. If the string contains characters that aren't A, C, G, or T then it is invalid and you should signal an error.

For example:

```
"GATTACA" -> 'A': 3, 'C': 1, 'G': 1, 'T': 2
"INVALID" -> error
```
